tests/5.1/dispatch/test_dispatch_nocontext.c

```diff
@@ -0,0 +1,74 @@
+//===---- test_dispatch_nocontext.c -----------------------------------------===//
+// 
+// OpenMP API Version 5.1
+//
+// Uses dispatch construct as context for variant directive. Uses nocontext clause
+// which can determine whether dispatch is a viable construct for the variant directive.
+// When nocontext is true, then dispatch is not a viable construct and vice versa. 
+//
+// Inspired by "OpenMP 5.1 Features: The Dispatch Construct" video:
+// https://www.youtube.com/watch?v=ruugaX95gIs
+// 
+//===-------------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include "ompvv.h"
+#include <stdbool.h>
+
+#define N 1024
+
+int arr[N]; // implicit map array 
+int errors;
+int i = 0;
+
+void add_two(int *arr);
+
+#pragma omp declare variant(add_two) match(construct={dispatch})
+void add(int *arr){
+   for (int i = 0; i < N; i++){ // Base function adds 1 to array values
+      arr[i] = i+1;
+   }
+}
+
+void add_two(int *arr){
+   for (int i = 0; i < N; i++){
+      arr[i] = i+2; // Variant function adds 2 to array values
+   }
+}
+
+int test_wrapper() { 
+   errors = 0;
```

Comment:
Make this an array of size 2.

Reply:
Shouldn't this be size 3? That is: no dispatch, dispatch with nocontext(true) and with nocontext(false)?

Reply:
If there is no dispatch considered, that is not an error, so there is no need to keep track of it.